application-licensing-licensor/application-licensing-licensor-api/src/main/resources/templates/newVersionAvailable.vm

```diff
@@ -4,13 +4,29 @@
 #set ($smallIcon = "arrow_up")
 #set ($compositeEventDate = $escapetool.xml($datetool.whenIs($event.dates.get(0))))
 
+#macro (getExtensionUpgradeURL $event)
+  ## The event body contains information about the extension id, version and targeted namespace.
+  #set ($extensionInfo = $event.getBody().split(' - '))
```

Comment:
First of all, I think the message contains the extension name not id. Thus, what happens if the extension name contains ``' - '``? This is too fragile and thus error-prone. If you want to pass structured information in the message body then why not use a standard format such as JSON or XML? Or maybe you can extract this information from elsewhere.